### ISSUE TITLE
Failure to create opengl context in shader viewcontroller on iPhone 5 iOS 10

### ISSUE THREAD

**Issue by alhill10** [beautypi/shadertoy-iOS-v2#16]
Hi there,

I've just installed the app from the app store on my iphone and it is unable to create a GL context in thje shader viewcontroller. From a quick glance at the app code I'm guessing it may have something to do with the GL ES version. I'll dig around and try to find a fix.

**Comment by alhill10** [comment]
Yep, this is most likely the issue. OpenGL ES 3.0 support started with iPhone 5s https://developer.apple.com/library/content/documentation/DeviceInformation/Reference/iOSDeviceCompatibility/HardwareGPUInformation/HardwareGPUInformation.html#//apple_ref/doc/uid/TP40013599-CH106-SW1

Is it possible to support ES 2.0 as a fallback, or is shadertoy dependent on 3.0?

**Comment by Ninja-Koala** [comment]
I think this is a duplicate of issue 14:
https://github.com/beautypi/shadertoy-iOS-v2/issues/14

**Comment by reindernijhoff** [comment]
Hi,

Yes you're right. I had to switch from OpenGL ES 2.0 to OpenGL ES 3.0. That's because OpenGL ES 3.0 has (almost) the same syntax as WebGL 2 which is the new WebGL version used on Shadertoy.

I will have a look if it's possible to make an OpenGL ES 2.0 fallback (and automatically rewrite the WebGL 2 shaders to a WebGL 1 syntax).
